hazelcast/src/test/java/com/hazelcast/jet/SimpleTestInClusterSupport.java

```diff
@@ -126,12 +146,33 @@ public void supportAfter() {
             PacketFiltersUtil.resetPacketFiltersFrom(inst);
         }
         // after each test ditch all jobs and objects
-        List<Job> jobs = stillActiveInstances.get(0).getJet().getJobs();
-        SUPPORT_LOGGER.info("Ditching " + jobs.size() + " jobs in SimpleTestInClusterSupport.@After: " +
-                jobs.stream().map(j -> idToString(j.getId())).collect(joining(", ", "[", "]")));
-        for (Job job : jobs) {
-            ditchJob(job, instances());
+        try {
+            List<Job> jobs = stillActiveInstances.get(0).getJet().getJobs();
+            SUPPORT_LOGGER.info("Ditching " + jobs.size() + " jobs in SimpleTestInClusterSupport.@After: " +
+                    jobs.stream().map(j -> idToString(j.getId())).collect(joining(", ", "[", "]")));
+            for (Job job : jobs) {
+                ditchJob(job, instances());
+            }
+        } catch (RuntimeException maybeDestroyed) {
+            if (maybeDestroyed.getCause() instanceof DistributedObjectDestroyedException && isParallelTestExecution()) {
+                // Race condition between getJobs() and proxy destruction is possible
+                // when supportAfter() is invoked in parallel.
+                // Even if there are any remaining jobs they should be terminated later by cancelAllExecutions.
+                //
+                // Example exception:
+                // com.hazelcast.spi.exception.DistributedObjectDestroyedException:
+                // Proxy [hz:impl:mapService:__jet.records] was destroyed while being created.
+                //  This may result in incomplete cleanup of resources.
+                //  at com.hazelcast.jet.impl.JetInstanceImpl.getJobsInt(JetInstanceImpl.java:116)
```

Comment:
example of that issue in https://github.com/hazelcast/hazelcast/pull/26007#issuecomment-1819683879